### PR TITLE
Implement buildTargetScalaMainClasses.

### PR DESCRIPTION
**Pull request by magda-aug**
This code should be refactored - significant part of the logic is copied from the buildTargetScalacOptions method. 

The code currently lacks tests.

### CODE REVIEW

File: src/main/java/org/jetbrains/bsp/bazel/server/ScalaBspServer.java [hunk 1]
```diff
@@ -101,4 +121,18 @@ public ScalaBspServer(
...
+  private List<ScalaMainClass> collectMainClasses(
+      BuildTargetIdentifier target,
+      Map<String, List<String>> mainClassesNames,
+      Map<String, List<String>> targetsOptions) {
+    return mainClassesNames.getOrDefault(target.getUri(), Lists.newArrayList()).stream()
+        .map(
+            mainClassName ->
+                new ScalaMainClass(
+                    mainClassName,
+                    new ArrayList<>(),
```
Comment (agluszak) [inline src/main/java/org/jetbrains/bsp/bazel/server/ScalaBspServer.java:134]: nitpick: sometimes you use `Lists.newArrayList()` and sometimes `new ArrayList<>()`

**Comment by andrefmrocha** [comment]
Are we planning on not merging this until the testkit is merged? 

Should this still be drafted?

**Comment by agluszak** [comment]
> Are we planning on not merging this until the testkit is merged?
> 
> Should this still be drafted?

afaik yup, we're waiting on the testkit, cc @magda-aug

**Comment by magda-aug** [comment]
new version - https://github.com/JetBrains/bazel-bsp/pull/36

**Pull request closed without merging**